ada_2_fhir-r4/zibs2020/payload/0.0/zib-AddressInformation.xsl

```diff
@@ -0,0 +1,170 @@
+                                    <coding>
+                                        <system value="{normalize-space(adres_soort/@codeSystem)}"/>
+                                        <code value="HP"/>
+                                        <display value="Officieel adres"/>
+                                    </coding>
+                                </valueCodeableConcept>
+                            </extension>
+                            <use value="home"/>
+                        </xsl:when>
+                        <!-- Tijdelijk adres -->
+                        <xsl:when test="@code = 'TMP'">
+                            <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-zib-AddressInformation-AddressType">
+                                <valueCodeableConcept>
+                                    <coding>
+                                        <system value="{normalize-space(adres_soort/@codeSystem)}"/>
+                                        <code value="TMP"/>
+                                        <display value="Tijdelijk adres"/>
+                                    </coding>
+                                </valueCodeableConcept>
+                            </extension>
+                            <use value="temp"/>
+                        </xsl:when>
+                        <!-- Werkadres -->
+                        <xsl:when test="@code = 'WP'">
+                            <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-zib-AddressInformation-AddressType">
+                                <valueCodeableConcept>
+                                    <coding>
+                                        <system value="{normalize-space(adres_soort/@codeSystem)}"/>
+                                        <code value="WP"/>
+                                        <display value="Werkadres"/>
+                                    </coding>
+                                </valueCodeableConcept>
+                            </extension>
+                            <use value="work"/>
+                        </xsl:when>
+                        <!-- Vakantie adres -->
+                        <xsl:when test="@code = 'HV'">
+                            <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-zib-AddressInformation-AddressType">
+                                <valueCodeableConcept>
+                                    <coding>
+                                        <system value="{normalize-space(adres_soort/@codeSystem)}"/>
+                                        <code value="HV"/>
+                                        <display value="Vakantie adres"/>
+                                    </coding>
+                                </valueCodeableConcept>
+                            </extension>
+                        </xsl:when>
+                    </xsl:choose>
+                </xsl:for-each>
+                <xsl:if test="$lineItems">
```

Comment:
Waarom wordt dit eerst in een variabele geplaatst?

Reply:
Het line element bevat veel informatie en deze heb ik gespiekt bij STU3:

`    <xsl:variable name="lineItems" as="element()*">
                <xsl:for-each select="straat/@value">
                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-streetName">
                        <valueString value="{normalize-space(.)}"/>
                    </extension>
                </xsl:for-each>
                <xsl:for-each select="huisnummer/@value">
                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-houseNumber">
                        <valueString value="{normalize-space(.)}"/>
                    </extension>
                </xsl:for-each>
                <xsl:for-each select="huisnummerletter/@value | huisnummertoevoeging/@value">
                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-buildingNumberSuffix">
                        <valueString value="{normalize-space(.)}"/>
                    </extension>
                </xsl:for-each>
                <xsl:for-each select="additionele_informatie/@value">
                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-unitID">
                        <valueString value="{normalize-space(.)}"/>
                    </extension>
                </xsl:for-each>
                <xsl:for-each select="aanduiding_bij_nummer/@value">
                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-additionalLocator">
                        <valueString value="{normalize-space(.)}"/>
                    </extension>
                </xsl:for-each>
            </xsl:variable>`
            

Vervolgens wordt er in het adres element de variable geplaats. Dit is beter qua leesbaarheid en onderhoudbaarheid. 

` <xsl:if test="$lineItems">
                    <line>
                        <xsl:copy-of select="$lineItems"/>
                    </line>
                </xsl:if>`

Reply:
Moved the $LineItems logic above the line element